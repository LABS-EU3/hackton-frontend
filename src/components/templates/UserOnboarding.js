import React, { useEffect } from "react";
import { Redirect, useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import queryString from "query-string";
import { socialAuthLoad, verifyEmail } from "../../store/user/actions";
import { FormLayout, Header, Footer } from "../organisms/index";

const UserOnboarding = ({
  ctaText,
  imageType,
  imageText,
  formHeader,
  formParagraph
}) => {
  let { search } = useLocation();
  const dispatch = useDispatch();
  const { token } = useSelector(state => state.currentUser);
  const { state } = useLocation();

  useEffect(() => {
    const { google, github, verified } = queryString.parse(search);
    if (google || github) {
      dispatch(socialAuthLoad());
    }
    if (verified) {
      dispatch(verifyEmail());
    }
  }, [search, dispatch]);

  if (token) {
    return <Redirect to={state?.from || '/dashboard'} />;
  }
  return (
    <div>
      <Header />
      <FormLayout
        ctaText={ctaText}
        imageType={imageType}
        imageText={imageText}
        formHeader={formHeader}
        formParagraph={formParagraph}
      />
      <Footer />
    </div>
  );
};

export default UserOnboarding;
