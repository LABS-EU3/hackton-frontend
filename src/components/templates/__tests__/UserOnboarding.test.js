import React from 'react'
import { Router, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react'
import UserOnboarding from '../UserOnboarding';
const history = createMemoryHistory()

const renderWithRouter = Component => render(
  <Router history={history}>
    <Route component={Component} />
  </Router>
)

describe("It should render <UserOnboarding/> template correctly", () => {
  
    it("renders the User onboarding templatet correctly", () => {
      const template = () => renderWithRouter(UserOnboarding)
      expect(template).toMatchSnapshot();
    });
  });