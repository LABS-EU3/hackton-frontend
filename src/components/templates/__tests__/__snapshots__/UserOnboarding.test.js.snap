// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render <UserOnboarding/> template correctly renders the User onboarding template correctly 1`] = `[Function]`;
